### DISCUSSION TITLE
Adding theme preload causes hydration error

### DISCUSSION THREAD

**Discussion by bostjanpisler** [wakujs/waku#1038] [Q&A] [answered]
Adding the theme preload logic form this example causes a hydration error, I'm using the base create starter, what code am I missing from the example? 

https://github.com/JesseKoldewijn/waku-tw4-shadcn-starter/blob/main/src/middleware/themePreload.ts

Error:
A tree hydrated but some attributes of the server rendered HTML didn't match the client properties. This won't be patched up. This can happen if a SSR-ed Client Component used:

**Comment by tylersayshi** [comment, marked as answer]
middleware that changes the rendered result before it reaches the client will cause a hydration error

you could use suppressHydrationError in this case since you are deliberately making the change between the client and server: https://react.dev/reference/react-dom/client/hydrateRoot#suppressing-unavoidable-hydration-mismatch-errors

It might be better though to set the theme preload in your RSC with `unstable_getHeaders` instead of needing the middleware

```ts
import { unstable_getHeaders } from 'waku/server';

...

<html lang={"${lang}"} class={themeFromHeaders}>
```

you'll need to use the root element api too: https://waku.gg/blog/root-element-api

hope this helps!

**Reply by bostjanpisler** [reply]
Yes, suppressHydrationError felt a bit hack-y. 

Thank you for the idea, I'll try with getHeaders.

**Reply by bostjanpisler** [reply]
The root element api did it, I missed it in the docs. Appreciate the help!
